org.osgi.test.common/src/main/java/org/osgi/test/common/annotation/config/InjectConfiguration.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.common.annotation.config;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/*
+ * Reads existing a Configuration from the Config-Admin
+ * Inject the `org.osgi.service.cm.Configuration` (if used on a Parameter or Field) for further updates or delete of the `org.osgi.service.cm.Configuration`
+ *
+ */
+@Inherited
+@Target({
+	PARAMETER, FIELD
+})
+
+@Retention(RUNTIME)
+@Documented
+
+public @interface InjectConfiguration {
+
+	static long DEFAULT_TIMEOUT = 200l;
+
+	/**
+	 * The pid of the Configuration.<br>
+	 * <br>
+	 * <ul>
+	 * <li><code>pid</code> - a service not a factory
+	 * <li><code>pid~factoryName</code> - a factory with given factoryName
+	 * </ul>
+	 *
+	 * @return The pid
+	 */
+	String value();
+
```

Comment:
Should we not also allow a `WithConfiguration` argument? So one can say

```java
@InjectConfiguration(@WithConfiguration(...))
```

Reply:
now you can use
@InjectConfiguration("pid")
@InjectConfiguration(withConfig=@WithConfiguration(...))
@InjectConfiguration(withFactoryConfig=@WithFactoryConfiguration(...))